docs/cookbook/create-basic-grid.md

```diff
@@ -0,0 +1,429 @@
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Component\Grid\Grid
+     */
+    public function create(): Grid
+    {
+-       /* @TODO: implement */
++       $grid = $this->gridFactory->create('salesmanGrid', $this->createAndGetDataSource());
+
++       return $grid;
+    }
+```
+
+Now, let's implement `createAndGetDataSource` method that should be in the same `SalesmanGridFactory` and will look like this
+```php
+    /**
+     * @return \Shopsys\FrameworkBundle\Component\Grid\DataSourceInterface
+     */
+    protected function createAndGetDataSource(): DataSourceInterface
+    {
+        $queryBuilder = $this->entityManager->createQueryBuilder();
+
+        $queryBuilder->select('s')
+            ->from(Salesman::class, 's');
+
+        return new QueryBuilderDataSource($queryBuilder, 's.id');
+    }
+```
+
+Do not forget to add import of `Shopsys\ShopBundle\Model\Salesman\Salesman` to `SalesmanGridFactory`.
+
+### 1.3 Add columns to the grid
+We prepared our grid, but for now, it is not rendered anywhere and does not contain any columns.
+We are going to change this now.
+
+First, we add columns we want to see into `create` method of `SalesmanGridFactory`
+
+```diff
+    public function create(): Grid
+    {
+        $grid = $this->gridFactory->create('salesmanGrid', $this->createAndGetDataSource());
+
++       $grid->addColumn('id', 's.id', t('Id'));
++       $grid->addColumn('name', 's.name', t('Name'));
++       $grid->addColumn('registeredAt', 's.registeredAt', t('Registered'));
+
+        return $grid;
+    }
+```
+
+*Note: In the example above, the column names are translated. Do not forget to [dump translations](/docs/introduction/console-commands-for-application-management-phing-targets.md#dump-translations).*
+
+## 2. Display the grid
+
+Grid is ready to show all the salesmen from the database.
+And we just need to render grid itself using a controller.
+
+### 2.1 Create a new controller & action
+We need to inject (pass through constructor) `SalesmanGridFactory` created earlier and pass grid view to the template.
+```php
+namespace Shopsys\ShopBundle\Controller\Admin;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Shopsys\FrameworkBundle\Controller\Admin\AdminBaseController;
+use Shopsys\ShopBundle\Grid\Salesman\SalesmanGridFactory;
+
+class SalesmanController extends AdminBaseController
+{
+    /**
+     * @var \Shopsys\ShopBundle\Grid\Salesman\SalesmanGridFactory
+     */
+    protected $salesmanGridFactory;
+
+    /**
+     * @Route("/salesman/list/")
+     */
+    public function listAction()
+    {
+        $grid = $this->salesmanGridFactory->create();
+
+        return $this->render('@ShopsysShop/Admin/Content/Salesman/list.html.twig', [
+            'gridView' => $grid->createView(),
+        ]);
+    }
+```
+
+### 2.2 Create a new template
+Finally, it is time to create a new twig template, `src/Shopsys/ShopBundle/Resources/views/Admin/Content/Salesman/list.html.twig` with the following content:
```

Comment:
on [shopsys/demoshop] we have it in `app/Resources` but we will se during tests

Reply:
AFAIK, you need to have it there when you are overriding a template from the framework. But in this case, we are adding new functionality, so it should not matter where the templates are, right?